netmiko/cisco/cisco_vxr_ssh.py

```diff
@@ -236,95 +319,186 @@ def send_command(self, command_string, expect_string=None,
+        started or since the last commit was made from this session. You can use the 'show
+        configuration commit changes' command to browse the changes.
+
+        Exit of configuration mode with pending changes will cause the changes to be discarded and
+        an exception to be generated.
+        """
+        if delay_factor is not None:
+            warnings.warn(DELAY_FACTOR_DEPR_SIMPLE_MSG, DeprecationWarning)
+            log.warning(DELAY_FACTOR_DEPR_SIMPLE_MSG)
+        commit_error_dialog_dict = kwargs.get('commit_error_dialog_dict')
+        if confirm and not confirm_delay:
+            raise ValueError("Invalid arguments supplied to XR commit")
+        if confirm_delay and not confirm:
+            raise ValueError("Invalid arguments supplied to XR commit")
+        if comment and confirm:
+            raise ValueError("Invalid arguments supplied to XR commit")
+
+        # wrap the comment in quotes
+        if comment:
+            if '"' in comment:
+                raise ValueError("Invalid comment contains double quote")
+            comment = '"{0}"'.format(comment)
+
+        label = str(label)
+        error_marker = 'Failed to'
+        alt_error_marker = 'One or more commits have occurred from other'
+
+        # Select proper command string based on arguments provided
+        if label:
+            if comment:
+                command_string = 'commit label {0} comment {1}'.format(label, comment)
+            elif confirm:
+                command_string = 'commit label {0} confirmed {1}'.format(label, str(confirm_delay))
             else:
-                prompt = self.base_prompt
-            search_pattern = re.escape(prompt.strip())
+                command_string = 'commit label {0}'.format(label)
+        elif confirm:
+            command_string = 'commit confirmed {0}'.format(str(confirm_delay))
+        elif comment:
+            command_string = 'commit comment {0}'.format(comment)
         else:
-            search_pattern = expect_string
-
-        if normalize:
-            command_string = self.normalize_cmd(command_string)
+            command_string = 'commit'
+        if force:
+            command_string = command_string.replace("commit", "commit force")
+        if best_effort:
+            command_string = command_string.replace("commit", "commit best-effort")
+        if replace:
+            command_string = command_string.replace("commit", "commit replace")
+        log.info("commit string is ", command_string)
 
-        time.sleep(delay_factor * loop_delay)
-        self.clear_buffer()
-        self.write_channel(command_string)
+        output = ''
+        if replace:
+            commit_replace_marker = "This commit will replace or remove the entire running configuration"
+            self.write_channel(self.normalize_cmd(command_string))
+            self.read_until_pattern(pattern=commit_replace_marker)
```

Comment:
Just for all read_until_pattern in def commit , i think it would be better to catch the error and report as commit failure and connection check is OK ..that means router is busted.